### DISCUSSION TITLE
IncludeBase for IProjectionExpression

### DISCUSSION THREAD

**Discussion by Isitar** [LuckyPennySoftware/AutoMapper#3844] [Ideas]
Since the new release suggests using CreateProjection instead of CreateMap for mappings only used for projections it would be nice to have IncludeBase on IProjectionExpression instead of IMappingExpression.

Our architecture always has two return models, one for list and one for detail requests. with IncludeBase we can include the base (list) in our detail vm without code duplication.

**Comment by lbargaoanu** [comment]
I see `IncludeBase` and `Include` as symmetrical. And you cannot get the polymorphism with `ProjectTo`. So I guess it only works in some cases. You have the same source type and you specify a concrete destination and that works? Very well :) But if it doesn't work in the general case, that's a good enough reason not no have it. cc @jbogard

**Comment by Isitar** [comment]
Do you have a better suggestion how to solve this scenario using CreateProjection?
I could use CreateProjection on the parent Class and then CreateMap on the child one i guess, so i benefit half the time from a speed-boost. Altough I don't know if it will work this way?

**Reply by lbargaoanu** [reply]
Instead of `IncludeBase` you could write an extension method with the common configuration and call it from the derived maps.
Another thing you can try is to use `CreateMap<SourceBase, DestinationBase>().Include<Source, Destination>()` and `CreateProjection<Source, Destination>()`. If it works, good for you, otherwise I don't know what to tell you :)
